### PR TITLE
Add basic calc test outline

### PR DESCRIPTION
**Pull request by SuperNerdEric**
I thought I'd add a really basic test outline so we can start building up tests for https://github.com/weirdgloop/osrs-dps-calc/issues/33

### CODE REVIEW

File: jest.config.ts [hunk 1]
```diff
@@ -5,6 +5,12 @@
 
 import type {Config} from 'jest';
 
+import nextJest from 'next/jest.js';
```
Comment (SuperNerdEric) [inline jest.config.ts:8]: There's something really funky with Nextjs/Jest together here that I really don't understand.

I found this solution from here (most recent reply): https://github.com/vercel/next.js/issues/40183
Here's another related issue in their Github: https://github.com/vercel/next.js/issues/58595

```
  ● Test suite failed to run

    Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.

    By default "node_modules" folder is ignored by transformers.

    Here's what you can do:
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • If you are trying to use TypeScript, see https://jestjs.io/docs/getting-started#using-typescript
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/configuration
    For information about custom transformations, see:
    https://jestjs.io/docs/code-transformation

    Details:

    C:\Users\Eric\WebstormProjects\osrs-dps-calc-wiki\node_modules\d3-array\src\index.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){export {default as bisect, bisectRight, bisectLeft, bisectCenter} from "./bisect.js";
                                                                                      ^^^^^^

    SyntaxError: Unexpected token 'export'

    > 1 | import {cross, max, sum} from "d3-array";
        | ^
      2 | import {HistogramEntry} from "@/types/State";
      3 |
      4 | export type HitTransformer = (hitsplat: number) => HitDistribution;

      at Runtime.createScriptFromCode (node_modules/jest-runtime/build/index.js:1505:14)
      at Object.<anonymous> (src/lib/HitDist.ts:1:1)
      at Object.<anonymous> (src/lib/CombatCalc.ts:3:1)
      at Object.<anonymous> (src/tests/utils/TestUtils.ts:4:1)
      at Object.<anonymous> (src/tests/CombatCalc.test.ts:3:1)`

File: src/tests/CombatCalc.test.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import {generateEmptyPlayer} from "@/state";
+import {calculate, getMonster} from "@/tests/utils/TestUtils";
+
+test('Empty player against Abyssal demon', () => {
+    let player = generateEmptyPlayer();
+    let monster = getMonster("Abyssal demon", "Standard");
+    let result = calculate(player, monster);
+
+    expect(result.maxHit).toBe(11);
+    expect(result.dps).toBe(0.6668319969138353);
```
Comment (SuperNerdEric) [inline src/tests/CombatCalc.test.ts:11]: Not sure on the level of precision we want in this. Currently it fails if it isn't exactly the same.

Comment (Nightfirecat) [reply src/tests/CombatCalc.test.ts:11]: [You can use `toBeCloseTo()` instead](https://jestjs.io/docs/expect#tobeclosetonumber-numdigits) so it doesn't have to be exactly identical. Given that [results have a fixed precision of 3 digits](https://github.com/weirdgloop/osrs-dps-calc/blob/f705dd355d406caa784eec24cf160c7dceb1de1c/src/app/components/results/ResultsContainer.tsx#L25), we'd probably be safe matching that. (if that approach is taken, let's export a var for this precision so it doesn't have to be hard-coded about)

**Comment by jayktaylor** [comment]
Thanks for your work on this! This is probably the most basic test of the actual calculator code that we can implement right now. Other tests will become more useful once we're confident enough with our formulae (there are still some numbers that may be slightly inaccurate right now in edge cases). We should also think about integrating the tests into our GitHub Actions CI, but we can action that as part of #33 too.
